### ISSUE TITLE
how to reset pull "u-pull-1/3@mobile" at desktop width ?

### ISSUE THREAD

**Issue by codemonkeynorth** [inuitcss/inuitcss#316]
Hi,

I'm trying to do a pull on mobile, and then remove that on desktop

I thought 0/3 might do it, but it doesn't seem to work
`u-pull-1/3@mobile u-pull-0/3@desktop u-1/3`

thanks for any advice... I'm not sure it's built in at the moment, unless I've missed something.
ie something like `u-pull-none@` , `u-pull-reset@` etc would be useful

this works for specific >0 push/pull amount, so i know it's not my setup
`u-pull-1/3@mobile u-push-1/3@desktop u-1/3`

thanks
J

**Comment by herzinger** [comment]
Good point... I never use pull/push, so it's the first time I think about it, but we don't have anything of the sort. The pull/push utilities are generated based on the same code we use for the width utilities, and there's no 0 width. This is something we should include. For now, here is a quick and dirty fix (untested, but should do the trick):

```scss
@if (variable-exists(mq-breakpoints)) {

  @each $inuit-bp-name, $inuit-bp-value in $mq-breakpoints {

    @include mq($from: $inuit-bp-name) {
      
      @if ($inuit-offsets == true) {

        .u-push-reset#{$breakpoint} {
          left: auto !important;
        }

        .u-pull-reset#{$breakpoint} {
          right: auto !important;
        }

      }

    }

  }

}
```

Thanks for the feedback

**Comment by codemonkeynorth** [comment]
thanks for the quick feedback! that's done it i think.

i'd originally  tried changing the 1 to 0 in
`@for $numerator from 1 through $denominator {`

but i think `reset` makes far more sense than `0/3` obviously!

so now this works:
`o-layout__item u-pull-1/3@mobile u-pull-reset@tablet u-push-1/3@desktop u-1/3`

regards
J

**Comment by herzinger** [comment]
I think resetting to the natural state (`auto`) makes more sense than forcing `0`. But as I said, I don't use those classes, so maybe there's a use case where the other way around is preferable. I can't really think of any, anyway.

**Comment by florianbouvot** [comment]
@codemonkeynorth thanks to point out this issue.

@herzinger thank you, I create a first set of classes based on your comment but using `none` naming.

I'm not sure about removing non responsive reset classes, if someone don't follow mobile first approach, thoughts ?

**Comment by herzinger** [comment]
Our grid and our implementation of `Sass-MQ` is mobile first already anyway, so I don't think that's a concern. But I do think we need a non responsive set, since responsiveness is optional in the framework.

**Comment by florianbouvot** [comment]
@herzinger 

>But I do think we need a non responsive set, since responsiveness is optional in the framework.

If I understand your sentence well: you think we need non responsive pull/push ?

**Comment by herzinger** [comment]
Well, yes. Don't we have it already? The layout object and width utilities work both with or without breakpoint variants, so why would we ship pull/push with only responsive support? And that being the case, this reset should be available both ways too. That was my rationale.

That being said, having thought about that a little more, in a non-responsive environment there shouldn't be any need for resetting pull/push: simply removing the original pull/push utility class would be better.

TL;DR: 
1. We do need responsive pull/push if we don't have it already (I'm pretty sure we do);
2. No, we don't need the reset/none variant without breakpoints, since there's no reasonable use case for that as far as I can tell.

**Comment by codemonkeynorth** [comment]
2) yes. reset is only necessary if it's been previously pulled/push and that's only going to happen if there's a previous breakpoint. 

(I guess if javascript is involved in the layout changes you could obviously just removeClass on the push... no need to override with a reset... only caveat to that is you'd need to know which class to remove. a "force reset" class might be useful, but I guess you would do that with the js anyway. not something I've ever needed to do anyway)

I think my requirement has come from using the previous version of inuit. I think the responsive classes `--palm` etc were *just* for that breakpoint not above (for which you eg had `--palm-and-up`) , so i never had to reset the push/pull before because they weren't applied above that single breakpoint.

**Comment by herzinger** [comment]
Yes, since moving to `Sass-MQ` we adopted a mobile-first approach. About the js scenario you described, if you're not sure about the direction you should remove both classes. There is no caveat for declaring a removal of a class that already doesn't exist.

**Comment by codemonkeynorth** [comment]
Sorry yes it's a javascript application issue, no real need for css to cover it and mostly irrelevant here

given this to begin with
`<div class="o-layout__item u-push-1/3@mobile  u-push-2/3@desktop">`

i just meant having an override `addClass('u-push-reset')` might be useful for a specific state, but it's an edge case that can be dealt with in other ways via javascript anyway.
(removing the push classes is of course the correct approach, it just takes a little more parsing than adding an override!)

thanks
J

**Comment by herzinger** [comment]
>[...] but it's an edge case that can be dealt with in other ways via javascript anyway

That's the main point. If you are lazy, or don't have the knowledge to write a simple parser to remove the classes with javascript, you can always take the filthy, dirty road and add a overall reset class on your own project anyway, but it's not a common enough use case to afford the inclusion on the framework.

As a bonus, here is a quick way to do something like that in jQuery, using the optional function parameter of `removeClass()`:

```js
    $('[class*="u-push"], [class*="u-pull"]') // just showcasing the attribute selectors, but it could be anything
    .removeClass(function (i, className) {
  
    return (className.match (/\u-[push|pull]\S+/g) || []).join(' ');
    });
```

**Comment by csshugs** [comment]
To bring this topic back on track: Shouldn't we tackle this issue by incorporating the [solution @florianbouvot already worked out](https://github.com/inuitcss/inuitcss/compare/develop...tkt-0316)? IMO this is no “big” addition but will probably help a lot of people. I really think this is definitely worth a PR.

**Comment by csshugs** [comment]
@inuitcss/core what do you think about [this](https://github.com/inuitcss/inuitcss/compare/develop...tkt-0316) approach?

**Comment by nenadjelovac** [comment]
@csshugs 
>IMO this is no “big” addition but will probably help a lot of people. I really think this is definitely worth a PR.

I agree.

**Comment by csshugs** [comment]
#341 is merged.
